pkg/k8s/watchers/cilium_clusterwide_envoy_config.go

```diff
@@ -113,6 +113,14 @@ func (k *K8sWatcher) addCiliumClusterwideEnvoyConfig(ccec *cilium_v2.CiliumClust
 		return err
 	}
 
+	if len(resources.Listeners) > 0 {
+		// TODO: Policy does not need to be recomputed for this, but if we do not 'force'
+		// the bpf maps are not updated with the new proxy ports either. Move from the
+		// simple boolean to an enum that can more selectively skip regeneration steps (like
+		// we do for the datapath recompilations already?)
+		k.policyManager.TriggerPolicyUpdates(true, "Envoy Listeners added")
```

Comment:
Do you intend to resolve this in this PR? Otherwise, we should create a GH issue for follow ups and put this item there, while referencing the GH issue in the code comment.

Reply:
Also, I'm not following "Policy does not need to be recomputed for this, but if we do not 'force' the bpf maps are not updated with the new proxy ports either". AFAIU, policy needs to be recomputed in order for there to be a change to update in the BPF maps, so this `TriggerPolicyUpdates` sounds like it is needed in contrast to what the comment says :slightly_smiling_face:.

Reply:
Not planning for this PR, as this related also to other cases where policy regeneration is forced. In this particular case we'd only need the existing endpoint policy be synced to bpf policy maps, as that is where the procy redirection port is resolved. This is why the endpoint policy itself would not need to be recomputed.

Reply:
> In this particular case we'd only need the existing endpoint policy be synced to bpf policy maps, as that is where the procy redirection port is resolved.

So there are no BPF policymap updates needed here, but the proxy redirection port logic is embedded somewhere in the policy regeneration / recalculation path? And we should only be triggering that and not a full regeneration / recalculation? How difficult is it to pull that out, just out of curiosity?

Reply:
BPF policy maps need to be updated so that the previously skipped redirect rules are pushed to the bpf maps (see your first comment). We would need to push more context info with the regeneration trigger to be able to skip parts of the regeneration that are known to not be needed. We already do something similar to the BPF code generation, where we can skip compilation and only load the new programs, etc.